update CI job, packaging scripts and directory layout for Swift 3.1

jpsim opened the pull request.

tgoyne commented:
Oh, actually those look like real test failures...

jpsim commented:
Yeah, the crash in `testDefaultValuesFromNSNull` is concerning, and the linker error for the static framework. I think the latter might be caused by Xcode 8.3 no longer supporting iOS 7 as a minimum target.

tgoyne commented:
I can't reproduce the testDefaultValuesFromNSNull failure locally, and it passes when building with address sanitizer. It sure would be nice if xctest didn't swallow the error...

jpsim commented:
I've updated the matrix to skip ios-static on Xcode 8.3 and changed the packaging script to always compile the ObjC framework with Xcode 8.2. @tgoyne @bdash could one of you please take another look?

tgoyne commented on build.sh:
```diff
@@ -174,6 +181,13 @@ build_combined() {
         sh build.sh binary-has-bitcode "$LIPO_OUTPUT"
     fi
+
+    if [[ "$module_name" == "Realm" ]]; then
```
Don't actually need this check.

tgoyne commented on build.sh:
```diff
@@ -130,6 +130,13 @@ build_combined() {
+    local previous_realm_xcode_version="$REALM_XCODE_VERSION"
+    local previous_realm_swift_version="$REALM_SWIFT_VERSION"
+
+    if [[ "$module_name" == "Realm" ]]; then
```
I'd be inclined to push this check inside workaround_rdar_31302382.

jpsim replied:
it's a local variable, so you're suggesting passing the module name to the function?

jpsim replied:
actually, I was considering just inlining `force_xcode82` and `workaround_rdar_31302382` into here since we only invoke it once now.
